src/Client.php

```diff
@@ -542,21 +559,67 @@ protected function send(array $data, array $tags = [])
      */
     protected function sendMessages(array $messages)
     {
+        if (is_null($this->socket)) {
+            $this->socket = $this->connect();
+        }
+        $this->message = implode("\n", $messages);
+        $this->write();
+
+        return $this;
+    }
+
+    /**
+     * Attempt to write the current message to the socket
+     *
+     * @return bool
+     */
+    private function write()
```

Comment:
Please keep it protected. This function will always be there, so let user to override it if they want.